portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,72 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+private ArrayList<String> messages;
+
+// Create ArrayList 
+    @Override
+    public void init() {
+        messages = new ArrayList<>();
+    }
+  
+  @Override 
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = request.getParameter("text-input");
+    long time = System.currentTimeMillis(); 
+    messages.add(text);
+    System.out.println(messages); 
+
+    Entity taskEntity = new Entity("Task");
+    taskEntity.setProperty("text", text);
+    taskEntity.setProperty("time", time);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(taskEntity);
+  }
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello I'm Sana!");
+
+    // Convert to JSON
+    String json = convertToJsonUsingGson(messages);
+
+    //Send response as JSON
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+    Query query = new Query("text").addSort("time", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService(); 
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<String> messages = new ArrayList<>(); 
```

Comment:
Should this be at the top of the method? it looks like line 63 uses the messages variable.

Reply:
Removed this line, there was really no need for it here again. Thanks for pointing that out :)